app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -79,11 +90,27 @@ import com.jerboa.ui.theme.JerboaTheme
 class JerboaApplication : Application() {
     private val database by lazy { AppDB.getDatabase(this) }
     val accountRepository by lazy { AccountRepository(database.accountDao()) }
-    val appSettingsRepository by lazy { AppSettingsRepository(database.appSettingsDao()) }
+    val appSettingsRepository by lazy {
+        AppSettingsRepository(database.appSettingsDao(), database.searchHistoryDao())
+    }
+    val searchHistoryRepository by lazy {
+        SearchHistoryRepository(database.searchHistoryDao(), database.appSettingsDao())
+    }
 }
 
 class MainActivity : AppCompatActivity() {
     private val siteViewModel by viewModels<SiteViewModel>()
+    private val communityViewModel by viewModels<CommunityViewModel>()
+    private val personProfileViewModel by viewModels<PersonProfileViewModel>()
+    private val inboxViewModel by viewModels<InboxViewModel>()
+    private val communityListViewModel by viewModels<CommunityListViewModel>() {
+        CommunityListViewModelFactory((application as JerboaApplication).searchHistoryRepository)
+    }
+    private val createPostViewModel by viewModels<CreatePostViewModel>()
+    private val commentReplyViewModel by viewModels<CommentReplyViewModel>()
+    private val commentEditViewModel by viewModels<CommentEditViewModel>()
+    private val postEditViewModel by viewModels<PostEditViewModel>()
+    private val createReportViewModel by viewModels<CreateReportViewModel>()
```

Comment:
Why are these added?

Reply:
Wonky rebase, thanks for catching. Removed.